pilot/pkg/model/cluster_local.go

```diff
@@ -95,7 +95,10 @@ func (c *clusterLocalProvider) onMeshUpdated(e *Environment) {
 	// Collect the cluster-local hosts.
 	hosts := make(ClusterLocalHosts, 0)
 	for _, serviceSettings := range e.Mesh().ServiceSettings {
-		if serviceSettings.Settings.ClusterLocal {
+		if serviceSettings == nil {
+			continue
+		}
+		if serviceSettings.Settings.GetClusterLocal() {
```

Comment:
nit: you could do `.GetSettings().GetClusterLocal()`

Reply:
Just for the other lines, I used `serviceSettings == nil` check

Reply:
Done.